### ISSUE TITLE
Parameterless constructor is always tried when creating fake classes, even if there isn't one

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1395]
Found while working on #1367. We always ask DynamicProxy to create a Fake class using its parameterless constructor, even if the class hasn't got one.
This is wasteful and can clutter up the error messages.

**Comment by blairconrad** [comment]
A naive fix for this issue changes our failure messages:

```diff
   Failed to create dummy of type FakeItEasy.Specs.DummyCreationSpecsBase+ClassWithRecursiveDependency:
     No Dummy Factory produced a result.
     It is not a Task.
     It is not a value type.

   Below is a list of reasons for failure per attempted constructor:
-    Constructor with signature () failed:
-      No usable default constructor was found on the type FakeItEasy.Specs.DummyCreationSpecsBase+ClassWithRecursiveDependency.
-      An exception of type Castle.DynamicProxy.InvalidProxyConstructorArgumentsException was caught during this call. Its message was:
-      Can not instantiate proxy of class: FakeItEasy.Specs.DummyCreationSpecsBase+ClassWithRecursiveDependency.
-      Could not find a parameterless constructor.

     The following constructors were not tried:
       (*FakeItEasy.Specs.DummyCreationSpecsBase+AnIntermediateClassInTheRecursiveDependencyChain)
```

I find it reads a little awkwardly when there's nothing between "Below is a list of reasons for failure per attempted constructor:" and "The following constructors were not tried:" (and one could also debate whether the latter should even be nested in the former. I don't think any of it's _wrong_, and it all hinges on whether you consider us having "attempted" a constructor that we ultimately couldn't resolve the arguments for.

Want me to rework the message while I'm here? If so, I'd consider

1. de-nesting "The following constructors were not tried"
2. maybe even removing the line "Below is a list of reasons for failure per attempted constructor:". Alternatively replacing "attemped" with "found" or "discovered" or "considered"

**Comment by thomaslevesque** [comment]
> and one could also debate whether the latter should even be nested in the former

I don't think it should be nested

> maybe even removing the line "Below is a list of reasons for failure per attempted constructor:".

I'd say remove it if no constructor was attempted

**Comment by blairconrad** [comment]
Thanks for the merge, @thomaslevesque!

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.8.0](https://github.com/FakeItEasy/FakeItEasy/releases/4.8.0).
